### PR TITLE
Remove plants#5

### LINKED ISSUE #5: Remove Plants
From my garden's show page, I have the ability to delete my plant.  When I select the delete button, I am returned to my garden's show page, and I no longer see that plant.  I see a flash message saying "goodbye dear plant"

### PR DESCRIPTION
**Pull request by asmolentzov**
User Story 5: Remove Plants
Adds ability for user to delete plants:
- Adds 'Remove Plant' button to garden show page
- Adds route and controller method for plant destroy
- Adds flash message for success
- Updates factories for plants and gardens so the names increment
- All tests passing
Closes #5

### CODE REVIEW

File: app/views/gardens/show.html.erb [hunk 1]
```diff
@@ -3,8 +3,13 @@
 <div class="plants">
   <ul>
     <% @garden.plants.each do |plant| %>
-      <li><%= plant.name %><br />
-      Watering Requirements: <%= plant.times_per_week.round(0) %> times/week</li>
+      <div id="plant-<%= plant.id %>"
+        <li>
+          <%= plant.name %><br />
+          Watering Requirements: <%= plant.times_per_week.round(0) %> times/week<br />
+          <%= button_to 'Remove Plant', plant_path(plant), method: :delete %>
```
Comment (bendelonlee) [inline app/views/gardens/show.html.erb]: Here's an example of what I mean `button_to("Delete Item", dashboard_item_path(item), method: :delete, data: {confirm: "Are you sure?"})`

File: app/controllers/plants_controller.rb [hunk 2]
```diff
@@ -17,6 +17,14 @@ def create
     end
   end
   
+  def destroy
+    plant = Plant.find(params[:id])
+    garden = plant.garden
+    plant.destroy
+    flash[:success] = 'Goodbye dear plant'
```
Comment (abenetka) [inline app/controllers/plants_controller.rb]: I appreciate this flash message

File: app/views/gardens/show.html.erb [hunk 3]
```diff
@@ -3,8 +3,13 @@
 <div class="plants">
   <ul>
     <% @garden.plants.each do |plant| %>
-      <li><%= plant.name %><br />
-      Watering Requirements: <%= plant.times_per_week.round(0) %> times/week</li>
+      <div id="plant-<%= plant.id %>"
```
Comment (abenetka) [inline app/views/gardens/show.html.erb]: This is will be a conflict with the garden show page tests, but will be easy to resolve

File: spec/factories/gardens.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 FactoryBot.define do
   factory :garden do
     zip_code { "MyString" }
-    name { "MyString" }
-    user { nil }
+    sequence(:name) { |n| "Garden #{n}" }
```
Comment (abenetka) [inline spec/factories/gardens.rb]: This is much better than "MyString", thanks for updating

File: app/views/gardens/show.html.erb (hunk 1 above)
Comment (asmolentzov) [reply app/views/gardens/show.html.erb]: Added, thanks!

File: app/views/gardens/show.html.erb [hunk 5]
```diff
@@ -7,7 +7,7 @@
         <li>
           <%= plant.name %><br />
           Watering Requirements: <%= plant.times_per_week.round(0) %> times/week<br />
-          <%= button_to 'Remove Plant', plant_path(plant), method: :delete %>
+          <%= button_to 'Remove Plant', plant_path(plant), method: :delete, data: {confirm: "Are you sure?"} %>
```
Comment (abenetka) [inline app/views/gardens/show.html.erb]: 👍

**Pull request merged**